### PR TITLE
ci: migrate saucelabs bazel job to Github Actions

### PR DESCRIPTION
**Pull request by josephperrott**
Use Github Actions to run saucelabs bazel jobs

### CODE REVIEW

**Comment by josephperrott** [comment]
Example of successful run: https://github.com/angular/angular/actions/runs/6342924498/job/17229682527

File: tools/saucelabs-daemon/launcher/launcher.ts [hunk 1]
```diff
@@ -40,6 +40,8 @@ export function SaucelabsLauncher(
   this.on('start', (pageUrl: string) => {
     daemonConnection = createConnection({port: IPC_PORT}, () => _startBrowserTest(pageUrl, args));
 
+
+
```
Comment (devversion) [inline tools/saucelabs-daemon/launcher/launcher.ts:44]: leftover from debugging?

File: .github/workflows/ci-privileged.yml [hunk 2]
```diff
@@ -62,3 +62,29 @@ jobs:
...
+          # Checking out the pull request commit is intended here as we need to run the changed code tests.
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Install node modules
+        run: yarn install --frozen-lockfile
+      - name: Setup Bazel
+        uses: angular/dev-infra/github-actions/bazel/setup@1befe6d24fd5cab818e7f8bada40ab364781b8a8
+      - name: Setup Bazel Remote Caching
+        uses: angular/dev-infra/github-actions/bazel/configure-remote@ba9b4487ced515e5b4d87edd681a3bd9792444d6
+      - name: Set up Sauce Tunnel Daemon
+        run: yarn bazel run //tools/saucelabs-daemon/background-service -- 2 &
```
Comment (devversion) [inline .github/workflows/ci-privileged.yml:82]: We should keep this a visible variable and also use it down below when invoking Bazel `--jobs=`

```suggestion
        run: yarn bazel run //tools/saucelabs-daemon/background-service -- {{vars.BLA}} &
```

File: .github/workflows/ci-privileged.yml [hunk 3]
```diff
@@ -62,3 +62,29 @@ jobs:
...
+        env:
+          SAUCE_TUNNEL_IDENTIFIER: angular-framework-${{ github.run_number }}
+          SAUCE_USERNAME: ${{ vars.SAUCE_USERNAME }}
+          SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_ACCESS_KEY }}
+      - name: Run all saucelabs bazel tests
+        run: |
+          TESTS=$(./node_modules/.bin/bazelisk query --output label '(kind(karma_web_test, ...) intersect attr("tags", "saucelabs", ...)) except attr("tags", "fixme-saucelabs", ...)')
+          yarn bazel test --cache_test_results=no --config=saucelabs --jobs=2 ${TESTS}
```
Comment (devversion) [inline .github/workflows/ci-privileged.yml:90]: Seems like a testing leftover to disable caching?

File: .circleci/config.yml [hunk 4]
```diff
@@ -199,20 +123,6 @@ commands:
           # Yarn's requests sometimes take more than 10mins to complete (especially on Windows).
           no_output_timeout: 45m
 
-  notify_webhook_on_fail:
```
Comment (devversion) [inline .circleci/config.yml:202]: While are at it: Did we loose the Slack notifications for failed AIO deployments?

File: .circleci/config.yml [hunk 5]
```diff
@@ -26,44 +26,19 @@ version: 2.1
...
-      ignore:
-        - /pull\/\d+/
-
-# Filter to run a job on builds for the main branch only.
-var_8: &only_on_main_branch
-  filters:
-    branches:
-      only:
-        - main
+var_3: &workspace_location ~/
```
Comment (devversion) [inline .circleci/config.yml:38]: There are many more changes in here, unrelated to Saucelabs. Can these be in a separate commit w/ some explanation on e.g. "cleaning up things"?

File: tools/defaults.bzl [hunk 6]
```diff
@@ -338,6 +338,10 @@ def karma_web_test_suite(
             tags = tags + [
                 "manual",
                 "no-remote-exec",
+                # Requires network to be able to access saucelabs daemon
+                "requires-network",
+                # Prevent the sandbox from being used so that it can communicate with the saucelabs daemon
+                "no-sandbox",
```
Comment (devversion) [inline tools/defaults.bzl:344]: LGTM, out of curiosity: do you recall why this came up? I wonder how this worked currently in CircleCI because I'd expect the test to run within the linux k8 sandbox then.

Comment (josephperrott) [reply tools/defaults.bzl:344]: It was failing to be able to access the other socket between jobs, on Github Actions. I was able to reproduce the same on my linux machine.

My thought is that it has to do with the fact that it was runner in a docker container on CircleCI but here is in a full linux VM.

As I investigated I found that the sandbox was the "cause" of it.

File: .circleci/config.yml (hunk 5 above)
Comment (josephperrott) [reply .circleci/config.yml:38]: I will do the cleanup in a follow up PR.

File: .circleci/config.yml (hunk 4 above)
Comment (josephperrott) [reply .circleci/config.yml]: Created https://github.com/angular/angular/pull/51956

File: tools/defaults.bzl (hunk 6 above)
Comment (devversion) [reply tools/defaults.bzl:344]: Yeah, no worries to investigate more. But super surprising. I'd expect the sandbox network isolation to still work in a Docker container.

**Review by devversion — APPROVED** [review]

**Comment by alxhub** [comment]
This PR was merged into the repository by commit f93352481b58ba19ec0085a29421ffea83b0744e.
